authors/apps/authentication/backends.py

```diff
@@ -1,10 +1,55 @@
-# import jwt
-#
-# from django.conf import settings
-#
-# from rest_framework import authentication, exceptions
-#
-# from .models import User
+import jwt
 
-"""Configure JWT Here"""
+from django.conf import settings
 
+from rest_framework import authentication, exceptions
+
+from .models import User
+
+class JWTAuthentication(authentication.BaseAuthentication):
+    """Token authentication using JWT."""
+
+    authentication_header_prefix = 'token'
+
+    def authenticate(self, request):
+        """Checks authorization on every request."""
+
+        request.user = None
+        auth_header = authentication.get_authorization_header(request).split()
+        
+        if not auth_header:
+            return None
```

Comment:
Raise an `AuthenticationFailed` exception for missing headers too.

Reply:
The method is called on each endpoint. If I raise an error on missing error, an error will be returned on the login and signup endpoints too which dont require jwt authentication